main/main.go

```diff
@@ -158,3 +169,36 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func restoreIBU(ctx context.Context, client client.Client, log *logr.Logger) error {
+	ibu := &lcav1alpha1.ImageBasedUpgrade{}
+	filePath := filepath.Join(utils.Host, utils.IBUFilePath)
+	if err := lcautils.ReadYamlOrJSONFile(filePath, ibu); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+	log.Info("IBU CR found, restoring ...")
+	if err := client.Delete(ctx, ibu); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	status := ibu.Status
```

Comment:
Why is this `status` variable necessary? Also what's the point of the `ibu.Status = status` assignment below?

Reply:
Ack, yes please add a comment

Reply:
Done. BTW, the good news is that this does restore all status fields as is, including the timestamps.